src/list.rs

```diff
@@ -339,64 +364,36 @@ impl ListenerSlab {
         Some(state)
     }
 
-    /// Notifies a number of entries, either normally or as an additional notification.
+    /// Notifies a number of listeners.
     #[cold]
-    pub(crate) fn notify(&mut self, count: usize, additional: bool) {
-        if additional {
-            self.notify_additional(count);
-        } else {
-            self.notify_unnotified(count);
-        }
-    }
-
-    /// Notifies a number of entries.
-    #[cold]
-    pub(crate) fn notify_unnotified(&mut self, mut n: usize) {
-        if n <= self.notified {
-            return;
-        }
-        n -= self.notified;
-
-        while n > 0 {
-            n -= 1;
-
-            // Notify the first unnotified entry.
-            match self.start {
-                None => break,
-                Some(e) => {
-                    // Get the entry and move the pointer forward.
-                    let e = &self.entries[e.get()];
-                    self.start = e.next.get();
-
-                    // Set the state of this entry to `Notified` and notify.
-                    let was_notified = e.notify(false);
-
-                    // Update the counter.
-                    self.notified += was_notified as usize;
-                }
+    pub(crate) fn notify(&mut self, mut n: usize, additional: bool) {
+        if !additional {
+            // Make sure we're not notifying more than we have.
+            if n <= self.notified {
+                return;
             }
+            n -= self.notified;
         }
-    }
 
-    /// Notifies a number of additional entries.
-    #[cold]
-    pub(crate) fn notify_additional(&mut self, mut n: usize) {
         while n > 0 {
```

Comment:
(maybe later) a `for` loop would be nicer here (but not necessary, so only if it wouldn't occur a performance penality (and that it hasn't been done yet probably indicates that it does incur a penality idk; if that gets tested, it warrants a comment))